RIOT-OS/RIOT
# Tracking: move CPU/CPU_MODEL to Makefile.features

### Contribution description

This pull request is a tracking pull request for moving the definition of `CPU` and `CPU_MODEL` to the board Makefile.features.

Moving these definitions is required to be parse `Makefile.dep` before `Makefile.include.

It is part of https://github.com/RIOT-OS/RIOT/issues/9913

This pull request is going to be split

* [ ] Makefile.features define new file
* Required cleanups
* [ ] CPU_MODEL defined in `arch/cortexm.inc.mk`
* [ ] TODO provide a test for everything

### Testing procedure

First compiling everything in the CI should still work. The number of jobs should stay the same.

I will also provide a way to verify that the `FEATURES_PROVIDED` do not change.

### Issues/PRs references

Tracking issue https://github.com/RIOT-OS/RIOT/issues/9913 on dependency handling order.
